packages/common/src/codegen/utils/parseSystem.ts

```diff
@@ -10,32 +12,34 @@ export function parseSystem(
   source: string,
   contractName: string,
 ): undefined | { contractType: "contract" | "abstract" } {
-  const ast = parse(source);
-  const contractNode = findContractNode(ast, contractName);
-  if (!contractNode) return;
+  const parser = Parser.create(LanguageFacts.latestVersion());
+  const root = parser.parseFileContents(source).createTreeCursor();
 
-  const contractType = contractNode.kind;
-  // skip libraries and interfaces
-  // we allow abstract systems here so that we can create system libraries from them but without deploying them
-  if (contractType !== "contract" && contractType !== "abstract") return;
+  const contractCursor = findContractOrInterfaceNode(root, contractName);
+  if (!contractCursor) return;
 
-  const isSystem = ((): boolean => {
-    // if using the System suffix, assume its a system
-    if (contractName.endsWith("System") && contractName !== baseSystemName) return true;
+  assertNonterminalNode(contractCursor.node);
+  if (contractCursor.node.kind !== "ContractDefinition") {
+    // it's an interface
```

Comment:
Why not use the positive check instead?
```suggestion
  if (contractCursor.node.kind == "InterfaceDefinition") {
```

Reply:
Hmm, this is good point. My thought was that because previously `findContractNode` would return nodes for libraries, interfaces and contracts, the negative predicate was the simplest.